### PR TITLE
iPhone X Compatibility

### LINKED ISSUE #8: iPhone X compatibility
Keyboard constraint needs to factor in the bottom layout guide length/height:

simulator screen shot - iphone x - 2017-10-10 at 14 50 46

### PR DESCRIPTION
**Pull request by markst**
Resolves #8

example

### CODE REVIEW

File: KeyboardConstraint/KeyboardAdjustConstraint.m [hunk 1]
```diff
@@ -49,9 +49,27 @@ - (void)_keyboardDidChangeVisible:(NSNotification *)notification
...
+    CGRect intersection = CGRectIntersection(windowRelativeFrame, endFrame);
+    CGFloat keyboardHeight = 0.0;
+
+    if (!CGRectIsNull(intersection)) {
+        keyboardHeight = intersection.size.height;
+    }
+
+    // If one of the constraint items is a layout guide; subtract it's length:
+    
+    if (notification.name == UIKeyboardWillShowNotification) {
```
Comment (markst) [inline KeyboardConstraint/KeyboardAdjustConstraint.m]: @Cbieniak what you recon about this?

**Comment by ezescaruli** [comment]
@markst This looks good to me 👍 

I thought about using the `safeAreaInsets` property of `UIView`, but that's only available in iOS 11+ so we could change to that once we stop supporting old iOS versions.

Is the constraint still working fine in iPhones other than the X after these changes?

**Comment by markst** [comment]
@ezescaruli seems to be. Jak is using this branch & we've not had any issues yet!

**Comment by markst** [comment]
🤔 that's a funny example I made... I guess it does the job

**Pull request merged**